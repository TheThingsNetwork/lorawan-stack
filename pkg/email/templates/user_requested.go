// Copyright Â© 2022 The Things Network Foundation, The Things Industries B.V.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package templates

import (
	"context"

	"go.thethings.network/lorawan-stack/v3/pkg/email"
	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
)

func init() {
	tmpl, err := email.NewTemplateFS(
		fsys, ttnpb.NotificationType_USER_REQUESTED,
		email.FSTemplate{
			SubjectTemplate:      "A new user has requested to join {{ .Network.Name }}",
			HTMLTemplateBaseFile: "base.html.tmpl",
			HTMLTemplateFile:     "user_requested.html.tmpl",
			TextTemplateFile:     "user_requested.txt.tmpl",
		},
	)
	if err != nil {
		panic(err)
	}
	email.RegisterTemplate(tmpl)
	email.RegisterNotification(ttnpb.NotificationType_USER_REQUESTED, &email.NotificationBuilder{
		EmailTemplateName: ttnpb.NotificationType_USER_REQUESTED,
		DataBuilder:       newUserRequestedData,
	})
}

func newUserRequestedData(_ context.Context, data email.NotificationTemplateData) (email.NotificationTemplateData, error) {
	var nData ttnpb.CreateUserRequest
	if err := data.Notification().GetData().UnmarshalTo(&nData); err != nil {
		return nil, err
	}
	return &UserRequestedData{
		NotificationTemplateData: data,
		CreateUserRequest:        &nData,
	}, nil
}

// UserRequestedData is the data for the user_requested email.
type UserRequestedData struct {
	email.NotificationTemplateData
	*ttnpb.CreateUserRequest
}
