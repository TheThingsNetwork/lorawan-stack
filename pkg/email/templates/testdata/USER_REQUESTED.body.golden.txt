Dear John Doe,

A new user with with ID foo-usr just registered on The Things Network.
Since user registration requires admin approval, you need to approve this user before they can login.

You can go to https://console.cloud.thethings.network/admin-panel/user-management/foo-usr to review this user in the Console.
